src/routes/solid-start/guides/actions-and-mutations.mdx

```diff
@@ -0,0 +1,272 @@
+
+```tsx {5} title="src/routes/index.tsx"
+import { action } from "@solidjs/router";
+import { db } from "~/lib/db";
+
+const addPost = action(async (formData: FormData) => {
+	"use server";
+	const title = formData.get("title");
+	await db.insert("posts").values({ title });
+}, "addPost");
+
+export default function Page() {
+	return (
+		<form action={addPost} method="post">
+			<input name="title" />
+			<button>Add Post</button>
+		</form>
+	);
+}
+```
+
+## Showing optimistic UI
+
+To update the UI before the server responds, you can use the [`useSubmission`](/solid-router/reference/data-apis/use-submission) primitive:
+
+1. Import `useSubmission` from `@solidjs/router`
+2. Call `useSubmission` with your action
+3. Use the `pending` and `input` properties to display optimistic UI
+
+```tsx {19} {28-30} title="src/routes/index.tsx"
+import { For, Show } from "solid-js";
+import { action, useSubmission, query, createAsync } from "@solidjs/router";
+
+const getPosts = query(async () => {
+	const posts = await fetch("https://my-api.com/blog");
+	return await posts.json();
+}, "posts");
+
+const addPost = action(async (formData: FormData) => {
+	const title = formData.get("title");
+	await fetch("https://my-api.com/posts", {
+		method: "POST",
+		body: JSON.stringify({ title }),
+	});
+}, "addPost");
+
+export default function Page() {
+	const posts = createAsync(() => getPosts());
+	const submission = useSubmission(addPost);
+	return (
+		<main>
+			<form action={addPost} method="post">
+				<input name="title" />
+				<button>Add Post</button>
+			</form>
+			<ul>
+				<For each={posts()}>{(post) => <li>{post.title}</li>}</For>
+				<Show when={submission.pending}>
+					{submission.input?.[0]?.get("title")?.toString()}
+				</Show>
+			</ul>
+		</main>
+	);
+}
+```
+
+<Callout type="info" title="Multiple Submissions">
+	If you want to display optimistic UI for multiple concurrent submissions, you can use the [`useSubmissions`](/solid-router/reference/data-apis/use-submissions) primitive.
+</Callout>
+
+## Calling an action manually
+
+If you don't want to use a `<form>` element, you can use the `useAction` primitive to manually call an action:
+
+1. Import `useAction` from `@solidjs/router`
+2. Call `useAction` with your action
+3. Use the returned function to trigger the action
+
+```tsx {13} {17} title="src/routes/index.tsx"
+import { createSignal } from "solid-js";
+import { action, useAction } from "@solidjs/router";
+
+const addPost = action(async (title: string) => {
+	await fetch("https://my-api.com/posts", {
+		method: "POST",
+		body: JSON.stringify({ title }),
+	});
+}, "addPost");
+
+export default function Page() {
+	const [title, setTitle] = createSignal("");
+	const addPostAction = useAction(addPost);
+	return (
+		<div>
+			<input value={title()} onInput={(e) => setTitle(e.target.value)} />
+			<button onClick={() => addPostAction(title())}>Add Post</button>
```

Comment:
I would recommend adding a callout at the bottom of this section that explains why form actions are preferred over `useAction` in terms of accessibility and progressive enhancement.

Reply:
Like the Data Fetching guide, this guide is geared toward people who want to know "how to" do things, and not why and when.